E5_documentation.md

```diff
@@ -11,4 +14,45 @@ For doing so:
 - Check the documentation looks nice locally: you can generate the html files with ```make html``` (see requirements.txt).
 - push to a branch and perform a pull request to the main branch.
 
-After review and once the branch is merged into th main branch, the online documentation will be automatically updated.
\ No newline at end of file
+After review and once the branch is merged into th main branch, the online documentation will be automatically updated.
+
+## Documentation of all packages
+
+The software comprises several packages.
+These are the packages that are downloaded via treep when installing the software from source and using colcon as described [here](A1_overview_and_installation#via-colcon-workspace).
+Some of these package have documentation, which can be built locally
+
+### How to build
+
+Using colcon:
+
+```bash
+cd /path/to/Software
+cd workspace
+colcon build --cmake-args ' -DCMAKE_BUILD_TYPE=Release ' -DGENERATE_DOCUMENTATION=ON
+```
+
+For this to work, you may need to pip install some packages: m2r, mistune (version 0.8.4), recommonmark, sphinx, sphinxcontrib-moderncmakedomain, breath.
+
+Once compilation is successfully finished, html documentation can be found in paths:
+
+```
+/path/to/software/install/<package name>/share/<package name>/docs/
+```
+
+### How to have the documentation building for a package
+
+To have the documentation built for a package using the method above, this command has to be added to
+the ```CMakeLists.txt``` file of the package:
+
+```cmake
+add_documentation()
```

Comment:
It might be good to mention that this command is provided by the mpi_cmake_modules package (i.e. you also need `find_package(mpi_cmake_modules)`).